code/modules/atmospherics/machinery/portable/canister.dm

```diff
@@ -352,30 +393,72 @@
 /obj/machinery/portable_atmospherics/canister/attackby(obj/item/item, mob/user, params)
 	if(istype(item, /obj/item/stock_parts/cell))
 		var/obj/item/stock_parts/cell/active_cell = item
-		if(!cell_container_opened)
-			balloon_alert(user, "open the hatch first")
+		if(!panel_open)
+			balloon_alert(user, "open hatch first!")
 			return TRUE
 		if(!user.transferItemToLoc(active_cell, src))
 			return TRUE
 		if(internal_cell)
 			user.put_in_hands(internal_cell)
-			balloon_alert(user, "you successfully replace the cell")
+			balloon_alert(user, "you replace the cell")
 		else
-			balloon_alert(user, "you successfully install the cell")
+			balloon_alert(user, "you install the cell")
 		internal_cell = active_cell
 		return TRUE
+	if(is_wire_tool(item) && panel_open)
+		wires.interact(user)
+		return
+	if(istype(item, /obj/item/assembly_holder))
+		if(rig)
+			balloon_alert(user, "another device is in the way!")
+			return ..()
+		var/obj/item/assembly_holder/holder = item
+
+		user.balloon_alert_to_viewers("attaching rig...")
+		add_fingerprint(user)
+		if(!do_after(user, 2 SECONDS, target = src) || !user.transferItemToLoc(holder, src))
+			return
+		rig = holder
+		holder.master = src
+		holder.on_attach()
+		assemblies_overlay = holder
+		assemblies_overlay.layer = 3.3
+		assemblies_overlay.pixel_y -= 5
+		add_overlay(assemblies_overlay)
+		log_bomber(user, "attached [holder.name] to ", src)
+		last_rigger = user
+		user.balloon_alert_to_viewers("attached rig")
+		return
 
 	return ..()
 
+/obj/machinery/portable_atmospherics/canister/attack_hand_secondary(mob/user, list/modifiers)
+	if(!rig)
+		return
+	// mousetrap rigs only make sense if you can set them off, can't step on them
+	// If you see a mousetrap-rigged fuel tank, just leave it alone
+	rig.on_found()
+	if(QDELETED(src))
+		return
+	user.balloon_alert_to_viewers("detaching rig...")
+	if(!do_after(user, 2 SECONDS, target = src))
+		return
+	user.balloon_alert_to_viewers("detached rig")
+	user.log_message("detached [rig] from [src].", LOG_GAME)
+	if(!user.put_in_hands(rig))
+		rig.forceMove(get_turf(user))
+	rig = null
+	last_rigger = null
+	cut_overlays(assemblies_overlay)
```

Comment:
The bits clearing `rig` and `last_rigger` should be done in `Exited()` like this:
```
/obj/machinery/portable_atmospherics/canister/Exited(atom/movable/gone)
	. = ..()
	if(gone == rig)
		rig = null
		last_rigger = null
		cut_overlays(assemblies_overlay)
```
This method is also pretty sturdy against hard-dels caused by lingering references to the object.